V8.0.0 beta.0 (#17020)
mui/mui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

> ## 8.0.0-beta.0

Author: ```suggestion
## 8.0.0-beta.0

<img width="100%" alt="MUI X v8 Beta is live" src="https://github.com/user-attachments/assets/61ec4dd8-c946-456b-8b45-d51de8772f5d">

```

- 🎁 Add support for `@mui/material` version 7 in all X packages
- 🐞 Bugfixes
- 🌍 Improve Chinese (zh-CN), (zh-HK), (zh-TW), Czech (cs-CZ), Korean (ko-KR) and Slovak (sk-Sk) locales on the Data Grid
- 🌍 Improve Chinese (zh-CN), (zh-HK) and (zh-TW) locales on the Pickers

## Breaking changes

- ℹ️ The peer dependency on `@mui/material` has been updated to accept only v7.
  This has been done to increase the adoption rate of ESM.
  Since only v7 of `@mui/material` has proper ESM support, we decided to help with its adoption and fix numerous issues using X packages in environments where transpiling is not an option.

Special thanks go out to the community members for their valuable contributions:
@Blake-McCullough, @hlavacz, @k-rajat19, @layerok, @nusr, @owais635, @yelahj.
Following are all team members who have contributed to this release:
@alexfauquette, @arminmeh, @bernardobelchior, @cherniavskii, @DiegoAndai, @flaviendelangle, @Janpot, @JCQuintas, @KenanYusuf, @LukasTy, @MBilalShafi, @michelengelen, @noraleonte, @romgrk.

<!--/ HIGHLIGHT_ABOVE_SEPARATOR /-->

## Alpha release highlights

Below are the highlights of the alpha releases leading up to this beta release:

> - 🚀📊 **New Pro Chart**: It is now possible to create Funnel charts—perfect for visualizing conversions, sales pipelines and more!

Reviewer: Maybe we could group thos highlight by package and group / reword a bit.
For example the funnel and the radar wording are very different, they could probably be a single line.

Reviewer: Not sure, since we have some things shared.

I was just proposing a very basic cleaning of the list, something like:

- React 19 support
- Highlight Data Grid 1
- Highlight Data Grid 2
- ...
- Highlight Data Grid N
- Highlight Chart 1 
- etc...

Maybe with the same icon every time per product line (📊 for the charts)

And merge lines that can be merged to have a more readable list:
For example:

- 📊 New charts: [Funnel](/) (pro plan only) and [Radar(/)

Or if we want to keep two lines;

- 📊 New pro chart: [Funnel](/)
- 📊 New community chart: [Radar](/)

Author: WDYT about the updates: [Refactor alpha highlights](https://github.com/mui/mui-x/pull/17020/commits/e8449e4e3fb793298bb3bf379deb75ae9baf8507)? 🤔

Reviewer 2: I like this new strategy of using 📊 for all charts update :+1:

- 🎯 **Data Caching**: Improved [data caching](https://mui.com/x/react-data-grid/server-side-data/#data-caching). Check out our [recommendations](https://mui.com/x/react-data-grid/server-side-data/#improving-the-cache-hit-rate) for improving the cache hit rate.
- **Charts Composition**: `<ChartDataProvider />` and `<ChartsSurface />` components are now fully divided — [Learn more](https://next.mui.com/x/react-charts/composition/#overview).
- **Custom HTML Components**: Users can create their own HTML components using chart data — [Learn more](https://next.mui.com/x/react-charts/components/#html-components).
- 🔧 **Tooltip Customization**: Refactor Tooltip customization for charts — [Learn more](https://next.mui.com/x/react-charts/tooltip/#overriding-content).

> - ⚛️ **React 19 Support**.

Reviewer: Move that to the top?

### Data Grid

#### `@mui/x-data-grid@8.0.0-beta.0`

- [DataGrid] Add a slot for unsort icon in column menu (#16918) @layerok
- [DataGrid] Add click propagation and prevents default on `toggleMenu` click (#16845) @michelengelen
- [DataGrid] Anchor preference panel to columns/filter trigger (#16953) @KenanYusuf
- [DataGrid] Fix QuickFilter debounce overriding input value (#16856) @KenanYusuf
- [DataGrid] Fix `printOptions` not respecting `hideFooter` root prop (#14863) @k-rajat19
- [DataGrid] Fix `processRowUpdate()` error if the row is removed before it is executed (#16741) @arminmeh
- [DataGrid] Fix bug with adding and removing columns in active edit state (#16888) @Blake-McCullough
- [DataGrid] Fix columns update not restoring column definition defaults (#16970) @cherniavskii
- [DataGrid] Fix page scrolling when preference panel is opened (#17004) @KenanYusuf
- [DataGrid] Fix visual issue with pinned columns and row spanning (#16923) @MBilalShafi
- [DataGrid] Make column header menu button aria-labels unique (#16796) @owais635
- [DataGrid] Refactor: create base Pagination (#16759) @romgrk
- [DataGrid] Update CSS variable naming convention to singular (#16993) @KenanYusuf

> - [DataGrid] Use Material UI CSS vars (#16962) @KenanYusuf

Author: ```suggestion
- [DataGrid] Use Material UI CSS vars (#16962) @KenanYusuf
```

Same changes as in `@mui/x-date-pickers@8.0.0-beta.0`, plus:

- [DateRangeCalendar] Do not update the previewed day when hovering a day and the value is empty (#16892) @flaviendelangle
- [TimeRangePicker] Shift popper between start and end input on multi input field (#16920) @LukasTy

### Charts

#### Breaking changes

- Tick labels in the x-axis of cartesian charts will now have an ellipsis applied to prevent overflow.
  If your tick labels are being clipped sooner than you would like, you can increase the x-axis size by increasing its `height` property.
  The default line-height has also been changed to 1.25, so if you aren't customizing the line height for x-axis tick labels, make sure to double check if the result is desirable.

#### `@mui/x-charts@8.0.0-beta.0`

- [charts] Add axis highlight to the radar (#16868) @alexfauquette
- [charts] Add radar labels (#16839) @alexfauquette
- [charts] Allow breaking line for radar labels (#16947) @alexfauquette
- [charts] Allow circular grid on radar chart (#16870) @alexfauquette
- [charts] Allow customizing shape in scatter charts (#16640) @bernardobelchior
- [charts] Avoid spreading props in demos (#16857) @bernardobelchior

> - [charts] Fix React 18 tests failing due to missing `forwardRef` (#16894) @bernardobelchior

Reviewer 2: ```suggestion
- [charts] Fix React 18 tests failing due to missing `forwardRef` (#16894) @bernardobelchior
- [charts] Fix line highlight position with RTL (#16994) @alexfauquette
```

> - [charts] Fix React 18 tests failing due to missing `forwardRef` (#16894) @bernardobelchior

Author: We tend to keep internal changes if they make even the slightest sense. 🤔

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 8.0.0-beta.0

<img width="100%" alt="MUI X v8 Beta is live" src="https://github.com/user-attachments/assets/61ec4dd8-c946-456b-8b45-d51de8772f5d">

_Mar 18, 2025_

We'd like to offer a big thanks to the 22 contributors who made this release possible. Here are some highlights ✨:

> - 🚀 Add Time Range Picker component

Reviewer: Maybe add a link to the doc page?

Author: ```suggestion
- 🚀 Add [Time Range Picker](https://next.mui.com/x/react-date-pickers/time-range-picker/) component
```

- ⚛️ React 19 support.

- 🛠️ New and improved Data Grid [Toolbar component](https://next.mui.com/x/react-data-grid/components/toolbar/).
- 📦 Data Grid [data source](https://next.mui.com/x/react-data-grid/server-side-data/) is now available in the Community plan.
- 🚫 Add ["No columns" overlay](https://next.mui.com/x/react-data-grid/overlays/#no-columns-overlay) to Data Grid.
- 🍬 Improved design for Data Grid [Header filters](https://next.mui.com/x/react-data-grid/filtering/header-filters/).
- 🔄 Add Data Grid [Scroll restoration](https://next.mui.com/x/react-data-grid/scrolling/#scroll-restoration).
- 💫 Support [aggregation with server-side data](https://next.mui.com/x/react-data-grid/server-side-data/aggregation/).
- 🎁 Support [Server-side lazy loading](https://next.mui.com/x/react-data-grid/server-side-data/lazy-loading/) on the Data Grid.
- 🎯 Improved [data caching](https://next.mui.com/x/react-data-grid/server-side-data/#data-caching).
- 🏎️ Improve Data Grid aggregation, Excel export serialization, mount, resize and scrolling performance.
- 🎨 Improve Data Grid theming and add default background color.

- 📊 New Pro chart: [Funnel](https://next.mui.com/x/react-charts/funnel/).

> - 📊 New community chart: [Radar](https://next.mui.com/x/react-charts/radar/) is available in preview for testing.

Reviewer: ```suggestion
- 📊 New Community chart: [Radar](https://next.mui.com/x/react-charts/radar/) is available in preview for testing.
```

(or replace `Pro` with `pro` above)

- 📊 Charts legend is now an HTML element which can be styled more easily.
- 📊 Create [custom HTML components](https://next.mui.com/x/react-charts/components/#html-components) using chart data.
- 📊 Refactor Charts [Tooltip customization](https://next.mui.com/x/react-charts/tooltip/#overriding-content).
- 📊 Improve Charts [composition](https://next.mui.com/x/react-charts/composition/#overview).
- 📊 Charts support server-side rendering under [some conditions](https://next.mui.com/x/react-charts/getting-started/#server-side-rendering).

> - 📊 Decouple `margin` and `axis-size`: A new API to support multiple axes.

Reviewer: ```suggestion
- 📊 Add a new API to support multiple axes (decouple `margin` and `axis-size`)
```

Not sure @alexfauquette 
To focus on the feature, not the API

- 🔁 Support [automatic parents and children selection](https://next.mui.com/x/react-tree-view/rich-tree-view/selection/#automatic-parents-and-children-selection) for the Rich Tree View components.

### Data Grid

#### `@mui/x-data-grid@8.0.0-beta.0`

- [DataGrid] Add a slot for unsort icon in column menu (#16918) @layerok
- [DataGrid] Add click propagation and prevents default on `toggleMenu` click (#16845) @michelengelen
- [DataGrid] Anchor preference panel to columns/filter trigger (#16953) @KenanYusuf

> - [DataGrid] Fix QuickFilter debounce overriding input value (#16856) @KenanYusuf

Reviewer 3: ```suggestion
- [DataGrid] Fix `QuickFilter` debounce overriding input value (#16856) @KenanYusuf
```

Same changes as in `@mui/x-data-grid@8.0.0-beta.0`, plus:

- [DataGridPro] Fix header filters not displaying restored values (#16855) @MBilalShafi
- [DataGridPro] Fix infinite loading not reacting when scrolling to the end (#16926) @arminmeh

#### `@mui/x-data-grid-premium@8.0.0-beta.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')

Same changes as in `@mui/x-data-grid-pro@8.0.0-beta.0`, plus:

- [DataGridPremium] Fix selection propagation issues with controlled state (#16810) @MBilalShafi

### Date and Time Pickers

#### Breaking changes

- The `useClearableField` hook has been removed.
  The custom field component now receives the `clearable` and `onClear` props — [Learn more](https://next.mui.com/x/migration/migration-pickers-v7/#useclearablefield).
- The `ExportedUseClearableFieldProps`, `UseClearableFieldSlots`, `UseClearableFieldSlotProps`, and `UseClearableFieldResponse` types have been removed — [Learn more](https://next.mui.com/x/migration/migration-pickers-v7/#removed-types).

#### `@mui/x-date-pickers@8.0.0-beta.0`

- [l10n] Improve Chinese (zh-CN), (zh-HK) and (zh-TW) locales (#16966) @nusr

> - [pickers] Add `TimeRangePicker` component (#9431) @flaviendelangle

Reviewer: ```suggestion
- [pickers] Add the Time Range Picker component (#9431) @LukasTy and @flaviendelangle 
```

- [charts] Fix interaction performance (#16897) @JCQuintas
- [charts] Fix x-axis tick label overflow (#16709) @bernardobelchior
- [charts] Grid support time step below 1s (#16957) @alexfauquette
- [charts] Improve radar slice (#16932) @alexfauquette
- [charts] Radar add option to highlighting series (#16940) @alexfauquette
- [charts] Refactor zoom `isInteracting` behavior directly to community code (#16999) @JCQuintas
- [charts] Remove `fireEvent` usage from tests (#17006) @JCQuintas
- [charts] Remove dead voronoi code (#16886) @JCQuintas
- [charts] Remove the polar axis plugin from the default plugins of the ChartContainer (#16936) @alexfauquette
- [charts] Rename `useIsClient` (#16937) @bernardobelchior

#### `@mui/x-charts-pro@8.0.0-beta.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-charts@8.0.0-beta.0`.

### Tree View

#### `@mui/x-tree-view@8.0.0-beta.0`

Internal changes.

#### `@mui/x-tree-view-pro@8.0.0-beta.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')

Same changes as in `@mui/x-tree-view@8.0.0-beta.0`.

### Docs

> - [docs] Add `TimeRangePicker` to relevant validation demos (#16919) @LukasTy

Reviewer: ```suggestion
- [docs] Add the Time Range Picker to relevant validation demos (#16919) @LukasTy
```
